doc/nrf/gs_modifying.rst

```diff
@@ -84,54 +93,115 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
 Therefore, if you always use SES to maintain them, you do not need to worry about tagging.
-Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c` file as source::
+Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c` file as source:
+
+.. code-block:: c
 
    # NORDIC SDK APP START
    target_sources(app PRIVATE src/main.c)
    # NORDIC SDK APP END
 
+Advanced compiler settings
+==========================
+
+The application has full control over the build process.
+
+Using Zephyr's configuration options is the standard way of controlling how the system is built.
+These options can be found under Zephyr's menuconfig :guilabel:`Build and Link Features` > :guilabel:`Compiler Options`.
+For example, to optimize for size, select :kconfig:`CONFIG_SIZE_OPTIMIZATIONS`; to compile for speed, select :kconfig:`CONFIG_SPEED_OPTIMIZATIONS`.
+
+If you want to specify non-standard compiler options, pass them using :kconfig:`CONFIG_COMPILER_OPT`.
+In addition to Zephyr's configuration, you can decide which options are passed to CMake targets using CMake commands.
+
 .. _configure_application:
 
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+.. _configuration_system_overview:
 
-For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
+Configuration system overview
+=============================
 
-.. _configuring_vsc:
+Zephyr's :ref:`configuration system <zephyr:build_overview>` is distributed, and so is the |NCS| one.
+This means that there are multiple sources of configuration.
+In the |NCS|, the configuration system consists of the following elements:
 
-Configuring in the VS Code extension
-====================================
+* Just like in Zephyr, hardware-related options are handled through Devicetree specification (DTS) files.
+* Just like in Zephyr, software-related options are handled through Kconfig files.
+* Additionally for the |NCS|, memory layout configuration is handled through Partition Manager.
```

Comment:
We are wording that configuration systems should follow, so it will read strangely if we open with "Just like in Zephyr", no?

* Devicetree specification (DTS) files for hardware-related options.
* Kconfig files for software-related options.
* Partition Manager files for memory layout configuration, an |NCS| configuration system that is not available in Zephyr.